src/workloads/execution/locf.yml

```diff
@@ -0,0 +1,257 @@
+        allowDiskUse: true
+    - OperationMetricsName: TestStringWithSinglePartition
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{
+          $setWindowFields: {
+            sortBy: {_id: 1},
+            partitionBy: "$part1",
+            output: {
+                string: {
+                  $locf: "$string"}
+                  }
+          }
+        }]
+        cursor: {batchSize: *batchSize}
+        allowDiskUse: true
+    - OperationMetricsName: TestRandomTypeWithSinglePartition
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{
+          $setWindowFields: {
+            sortBy: {_id: 1},
+            partitionBy: "$part1",
+            output: {
+                randomType: {
+                  $locf: "$randomType"}
+                  }
+          }
+        }]
+        cursor: {batchSize: *batchSize}
+        allowDiskUse: true
+    - OperationMetricsName: TestNumericWithMultiplePartitions
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{
+          $setWindowFields: {
+            sortBy: {_id: 1},
+            partitionBy: {part1: "$part1", part2: "$part2"},
+            output: {
+                numeric: {
+                  $locf: "$numeric"}
+                  }
+          }
+        }]
+        cursor: {batchSize: *batchSize}
+        allowDiskUse: true
+    - OperationMetricsName: TestDateWithMultiplePartitions
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{
+          $setWindowFields: {
+            sortBy: {_id: 1},
+            partitionBy: {part1: "$part1", part2: "$part2"},
+            output: {
+                date: {
+                  $locf: "$date"}
+                  }
+          }
+        }]
+        cursor: {batchSize: *batchSize}
+        allowDiskUse: true
+    - OperationMetricsName: TestStringWithMultiplePartitions
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{
+          $setWindowFields: {
+            sortBy: {_id: 1},
+            partitionBy: {part1: "$part1", part2: "$part2"},
+            output: {
+                string: {
+                  $locf: "$string"}
+                  }
+          }
+        }]
+        cursor: {batchSize: *batchSize}
+        allowDiskUse: true
+    - OperationMetricsName: TestRandomTypeWithMultiplePartitions
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{
+          $setWindowFields: {
+            sortBy: {_id: 1},
+            partitionBy: {part1: "$part1", part2: "$part2"},
+            output: {
+                randomType: {
+                  $locf: "$randomType"}
+                  }
+          }
+        }]
+        cursor: {batchSize: *batchSize}
+        allowDiskUse: true
+
```

Comment:
Can you add a test or two with multiple outputs? Don't think we need all combinations, but one with all three types I think would be good to have.

Reply:
done